src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncSpillTests.cs

```diff
@@ -3789,9 +3789,11 @@ static async Task Main()
                 var comp = CreateCompilationWithMscorlibAndSpan(source, options: options);
                 comp.VerifyDiagnostics();
                 comp.VerifyEmitDiagnostics(
-                    // (9,66): error CS4007: 'await' cannot be used in an expression containing the type 'System.Span<int>'
-                    //         await Async1(F1(), G(F2(), stackalloc int[] { 1, 2, 3 }, await F3()));
-                    Diagnostic(ErrorCode.ERR_ByRefTypeAndAwait, "await F3()").WithArguments("System.Span<int>").WithLocation(9, 66)
+                    // (8,5): error CS4007: Instance of type 'System.Span<int>' cannot be preserved across 'await' or 'yield' boundary.
+                    //     {
+                    Diagnostic(ErrorCode.ERR_ByRefTypeAndAwait, @"{
+        await Async1(F1(), G(F2(), stackalloc int[] { 1, 2, 3 }, await F3()));
+    }").WithArguments("System.Span<int>").WithLocation(8, 5)
```

Comment:
Minor: Should the error location be `stackalloc int[] { 1, 2, 3 }` rather than the containing block?

Reply:
That does not look easily doable. The problematic local used across await is a synthesized local whose syntax is the `await F3()` expression and the node where the usage after await occurs has the whole block as syntax (that's what's used to report the diagnostic, it mostly is more useful than the former). In any case, we do not seem to have the location of the stackalloc expression at hand.